### PR TITLE
Gait selection develop

### PR DESCRIPTION
**Pull request by Olavhaasie**
## Description
This PR adds the new features made possible by the new gait selection :tada: 

## Changes
* Gait graphs
* Named idle positions in gait files
* Removed home gaits, since they are now generated
* Add small and large subgaits to walk in minimal for size transitioning
* Every subgait version on a new line for easier version control
* Removed the old airgait, test_versions and training gait directories since they were no longer up to date and never used

### CODE REVIEW

**Comment by gaiavdh** [comment]
The airgait, training and test_versions gait directories contain all the MARCH IV gaits. Are we sure we want to delete all the gaits they made? I think it could be nice for the new Movement Engineers to be able to take a look at what MARCH IV did. Not sure if anyone agrees on this? Or would it be a problem/inconvenient to keep them?

**Comment by Olavhaasie** [comment]
> The airgait, training and test_versions gait directories contain all the MARCH IV gaits. Are we sure we want to delete all the gaits they made? I think it could be nice for the new Movement Engineers to be able to take a look at what MARCH IV did. Not sure if anyone agrees on this? Or would it be a problem/inconvenient to keep them?

Okay I wasn't aware that these were still used. We can keep them, however, they won't work with the gait selection right away, since that requires some changes. I think these gait directories are never used as gait directory, so that shouldn't be a problem. I could re-add them in a separate directory called `march-iv-gaits` or something like that. Then they can still be used with the gait generator, but not with the gait selection.

**Comment by gaiavdh** [comment]
> > The airgait, training and test_versions gait directories contain all the MARCH IV gaits. Are we sure we want to delete all the gaits they made? I think it could be nice for the new Movement Engineers to be able to take a look at what MARCH IV did. Not sure if anyone agrees on this? Or would it be a problem/inconvenient to keep them?
> 
> Okay I wasn't aware that these were still used. We can keep them, however, they won't work with the gait selection right away, since that requires some changes. I think these gait directories are never used as gait directory, so that shouldn't be a problem. I could re-add them in a separate directory called `march-iv-gaits` or something like that. Then they can still be used with the gait generator, but not with the gait selection.

Yes that would be nice! I don't think it is necessary that they can be used with the gait selection, so just being able to open them in the gait generator should be fine.

**Comment by RutgerVanBeek** [comment]
Have you considered letting gait directories only differ by their yaml and not have all gaits copied in different directories. Some gait versions are in there a couple of times now, which is redundant I think. Furthermore if you copy a version to multiple directories and then only change when mistakes can slip in easily.

File: march_gait_files/airgait-v/default.yaml [hunk 1]
```diff
@@ -1,65 +1,223 @@
...
-                            left_close: MV_stairsdown_single_leftclose_v4}
-  stairs_down: {right_open: MV_stairsdown_rightopen_v9, left_swing: MV_stairsdown_leftswing_v9,
-                right_swing: MV_stairsdown_rightswing_v9, left_close: MV_stairsdown_leftclose_v9,
-                right_close: MV_stairsdown_rightclose_v9}
-  sofa_sit: {sit_down: MIV_final, sit_home: MIV_final}
-  sofa_stand: {prepare_stand_up: MIV_final, stand_up: MIV_final}
+  home_sit:
+    home_sit: MIV_final
+  home_stand:
+    home_stand: MIV_final
```
Comment (RutgerVanBeek) [inline march_gait_files/airgait-v/default.yaml]: These should be removed, right?

File: march_gait_files/test_versions-v/default.yaml [hunk 2]
```diff
@@ -1,64 +1,259 @@
...
-  tilted_path_right_straight_end: {right_open: MV_TP_right_straightend_rightopen_v5,
-                                   left_close: MV_TP_right_straightend_leftclose_v5}
-  tilted_path_right_knee_bend: {right_open: MV_TP_right_kneebend_rightopen_v5}
-  tilted_path_right_flexed_knee_step: {left_open: MV_TP_right_flexed_knee_step_leftopen_v1,
-                                       right_close: MV_TP_right_flexed_knee_step_rightclose_v1}
+  home_sit:
+    home_sit: MIV_ankle_2.5
+  home_stand:
+    home_stand: MIV_ankle_2.5
+  ramp_door_slope_up:
```
Comment (RutgerVanBeek) [inline march_gait_files/test_versions-v/default.yaml]: Here as well

File: march_gait_files/training-v/default.yaml [hunk 3]
```diff
@@ -1,71 +1,235 @@
...
-                               right_swing_1: MV_RD_slopedown_fixed_rightswing1_v1,
-                               left_swing_2: MV_RD_slopedown_fixed_leftswing2_v1,
-                               right_swing_2: MV_RD_slopedown_fixed_rightswing2_v1,
-                               left_close: MV_RD_slopedown_fixed_leftclose_v1}
-
-
+  home_sit:
+    home_sit: MIV_final
+  home_stand:
+    home_stand: MIV_final
```
Comment (RutgerVanBeek) [inline march_gait_files/training-v/default.yaml]: +1

**Review by RutgerVanBeek — CHANGES_REQUESTED** [review]
I think you should delete some home gaits.

**Comment by Olavhaasie** [comment]
> Have you considered letting gait directories only differ by their yaml and not have all gaits copied in different directories. Some gait versions are in there a couple of times now, which is redundant I think. Furthermore if you copy a version to multiple directories and then only change when mistakes can slip in easily.

I agree that would be better, however, the walk gait is different for minimal and the other gait directories, which I do not have a direct solution for.

**Comment by gaiavdh** [comment]
> > Have you considered letting gait directories only differ by their yaml and not have all gaits copied in different directories. Some gait versions are in there a couple of times now, which is redundant I think. Furthermore if you copy a version to multiple directories and then only change when mistakes can slip in easily.
> 
> I agree that would be better, however, the walk gait is different for minimal and the other gait directories, which I do not have a direct solution for.

The reason that the walk gait is different in the minimal gait directory is simply that we have not changed it there. So there's not really a reason for that and it is not a problem if it would be the same. However, one issue I see with only changing the default is that the different gait directories are also there for safety reasons. We make sure that in training-v are only gaits that are tested and can therefore be used with Sjaan. In test_versions-v are sometimes test gaits that we do not want to use on the exoskeleton. Alhtough I agree that it would be easier to have the gaits all in one directory, that would be an issue...

**Comment by RutgerVanBeek** [comment]
> The reason that the walk gait is different in the minimal gait directory is simply that we have not changed it there. So there's not really a reason for that and it is not a problem if it would be the same. However, one issue I see with only changing the default is that the different gait directories are also there for safety reasons. We make sure that in training-v are only gaits that are tested and can therefore be used with Sjaan. In test_versions-v are sometimes test gaits that we do not want to use on the exoskeleton. Alhtough I agree that it would be easier to have the gaits all in one directory, that would be an issue...

I would then extend the yaml to say which gaits are in the 'directory' or group as that is a better name in that case. You would still have to create a PR to add a gait to a directory. However it will be a PR of one line, so less would go wrong. Next to that, a version 4 stairs gait right open is the same in every directory/group.

**Review by RutgerVanBeek — APPROVED** [review]

**Review by Joris-JW — APPROVED** [review]

**Review by Roelemans — APPROVED** [review]

**Pull request merged**